### PR TITLE
refactor remoterx_

### PR DESCRIPTION
**Pull request by hsmade**

### CODE REVIEW

File: plugins/remoterx_ [hunk 1]
```diff
@@ -1,67 +1,80 @@
-#!/usr/bin/python
+#!/usr/bin/python2.7
```
Comment (hsmade) [inline plugins/remoterx_:1]: Always make sure to specify the version. Print in this case, for example is incompatible with python3

File: plugins/remoterx_ [hunk 2]
```diff
@@ -1,67 +1,80 @@
...
     else:
-        print 'state.value 0'
+        state = 0
+    print "state.value {}".format(state)
+
+    level = line.split("(")[1].split(")")[0]  # alternative is regexp, but this is more flexible
+    print "level.value {}".format(level)
 
-    print 'level.value ' + lines[bracketopen+2:bracketclose]
+if __name__ == "__main__":
```
Comment (hsmade) [inline plugins/remoterx_:79]: Best practice: Don't have any code executed if this program is not called directly, for example when importing.

File: plugins/remoterx_ (hunk 1 above)
Comment (realrolfje) [reply plugins/remoterx_:1]: Roger.

File: plugins/remoterx_ [hunk 3]
```diff
@@ -1,67 +1,80 @@
...
+        if char == '\n':
+            yield data[::-1], position  # note that we do not include the newline
+        data += char
+
+
+def trace_back(file_name, text):
+    """
+    Read a file backwards until we found `text`, then return the line that contains it
+
+    :param file_name: the file name to read
```
Comment (realrolfje) [inline plugins/remoterx_:37]: Documentation! Nice.

**Pull request merged**